### ISSUE TITLE
Dynamic API: Support command batching

### ISSUE THREAD

**Issue by mp911de** [mp911de/lettuce#432]
The dynamic API should support batching of commands when using synchronous and asynchronous executions.

```
@BatchSize(50)
interface MyCommands extends Commands {

  
  void set(String key, String value);

  Future<String> set(String key, String value);

  void set(String key, String value, CommandBatching commandBatching);
}

class CommandBatching {
    public static CommandBatching flush() {
        …
    }

    public static CommandBatching queue() {
        …
    }
}
```

Command batching for synchronous commands is only possible when the return type is ignored because the command result is only available after command execution. Reactive commands cannot participate in batching because of the deferred execution model.

Batching could be specified either by annotation `@BatchSize` or a `CommandBatching` that is specified on a per-invocation basis.

**Comment by mp911de** [comment]
Depends on #456
